.github/workflows/pull-db_test.yml

```
@@ -0,0 +1,260 @@
+        env:
+          GOPROXY: https://goproxy.io
+          GOSUMDB: sum.golang.org
+          TAGS: bindata
+      - name: run tests
+        run: timeout -s ABRT 50m make test-pgsql-migration test-pgsql
+        env:
+          GOPROXY: https://goproxy.io
+          TAGS: bindata gogit
+          RACE_ENABLED: true
+          TEST_TAGS: gogit
+          TEST_LDAP: 1
+          USE_REPO_TEST_DIR: 1
+
+  # SQLite Tests
+  db_sqlite_test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+      - name: setup go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '>=1.20.0'
+      - name: install dependencies
+        run: make deps-backend
+      - name: build
+        run: make backend
+        env:
+          GOPROXY: https://goproxy.io
+          GOSUMDB: sum.golang.org
+          TAGS: bindata gogit sqlite sqlite_unlock_notify
+      - name: run tests
+        run: timeout -s ABRT 50m make test-sqlite-migration test-sqlite
+        env:
+          GOPROXY: https://goproxy.io
+          TAGS: bindata gogit sqlite sqlite_unlock_notify
+          RACE_ENABLED: true
+          TEST_TAGS: gogit sqlite sqlite_unlock_notify
+          USE_REPO_TEST_DIR: 1
+
+  # Unit Tests
+  db_unit_tests:
+    runs-on: ubuntu-latest
+    services:
+      mysql:
+        image: mysql:5.7
+        env:
+          MYSQL_ALLOW_EMPTY_PASSWORD: yes
+          MYSQL_DATABASE: test
+        ports:
+          - "3306:3306"
+      elasticsearch:
+        image: elasticsearch:7.5.0
+        env:
+          discovery.type: single-node
+        ports:
+          - "9200:9200"
+      smtpimap:
+        image: tabascoterrier/docker-imap-devel:latest
+        ports:
+          - "25:25"
+          - "143:143"
+          - "587:587"
+          - "993:993"
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+      - name: setup go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '>=1.20.0'
+      - name: Add hosts to /etc/hosts
+        run: echo "127.0.0.1 mysql elasticsearch smtpimap" | sudo tee -a /etc/hosts
+      - name: install dependencies
+        run: make deps-backend
+      - name: build
+        run: make backend
+        env:
+          GOPROXY: https://goproxy.io
+          GOSUMDB: sum.golang.org
+          TAGS: bindata
+      - name: unit tests
+        run: make unit-test-coverage test-check
+        env:
+          GOPROXY: https://goproxy.io
+          TAGS: bindata
+          RACE_ENABLED: true
+          GITHUB_READ_TOKEN: ${{ secrets.GITHUB_READ_TOKEN }}
+      - name: unit tests (gogit)
+        run: make unit-test-coverage test-check
+        env:
+          GOPROXY: https://goproxy.io
+          TAGS: bindata gogit
+          RACE_ENABLED: true
+          GITHUB_READ_TOKEN: ${{ secrets.GITHUB_READ_TOKEN }}
+
+  # MySQL Tests
+  db_mysql_test:
+    runs-on: ubuntu-latest
+    services:
+      mysql:
+        image: mysql:5.7
```

```suggestion
        image: mysql:8.0
```

5.7 goes EOL [in 6 months](https://endoflife.date/mysql), may as well test on 8 now.

---

Thought I do see drone was testing 5.7 as well so we could defer this upgrade to later as it's not strictly related to the ci move.